lexos/processors/analyze/content_analysis_model.py

```diff
@@ -0,0 +1,461 @@
+
+        :return: error message if there is an empty str or None if there is
+                 no errors
+        """
+        error_msg = "Formula errors:<br>"
+        is_error = False
+        if self._formula.count("(") != self._formula.count(")"):
+            error_msg += "Mismatched parenthesis<br>"
+            is_error = True
+        if "sin()" in self._formula:
+            error_msg += "sin takes exactly one argument (0 given)<br>"
+            is_error = True
+        if "cos()" in self._formula:
+            error_msg += "cos takes exactly one argument (0 given)<br>"
+            is_error = True
+        if "tan()" in self._formula:
+            error_msg += "tan takes exactly one argument (0 given)<br>"
+            is_error = True
+        if "log()" in self._formula:
+            error_msg += "log takes exactly one argument (0 given)<br>"
+            is_error = True
+        if is_error:
+            return error_msg
+        return ""
+
+    def analyze(self) -> [str, str]:
+        """Generates html table containing counts, averages, & formula results
+
+        :return: formula_errors: str with error message or None if there is no
+                 errors.
+                 result_table: str containing result html table or None if
+                 there are any errors in the formula
+        """
+        self.count()
+        if self.is_secure():
+            formula_errors = self.save_formula()
+            self.generate_scores()
+            self.generate_averages()
+            result_table = self.to_html()
+        else:
+            formula_errors = "Formula error: Invalid input"
+            result_table = ""
+        return result_table, formula_errors
+
+    @property
+    def dictionaries(self) -> list:
+        return self._dictionaries
+
+    @dictionaries.setter
+    def dictionaries(self, dictionaries: list):
+        self._dictionaries = dictionaries
+
+    @property
+    def corpus(self) -> list:
+        return self._corpus
+
+    @property
+    def counters(self) -> list:
+        return self._counters
+
+    @property
+    def scores(self) -> list:
+        return self._scores
+
+    @property
+    def averages(self) -> list:
+        return self._averages
+
+    @property
+    def content_analysis_option(self) -> ContentAnalysisOption:
+        """Gets front-end options from ContentAnalysisReceiver
+
+        :return: front-end or test options
+        """
+        if self._test_options is not None:
+            if self._test_options.formula is not None:
+                self.save_formula()
+            return self._test_options
+        return ContentAnalysisReceiver().options_from_front_end()
+
+    @property
+    def toggle_all(self):
+        return self._toggle_all
+
+    @property
+    def test_option(self):
+        return self._test_options
+
+    @test_option.setter
+    def test_option(self, options):
+        self._test_options = options
+
+
+def count_phrases(dictionary, file):
```

Comment:
Now I know that counting phrases is a tricky thing to do, because we discussed it at length in our meeting. You should really be explaining exactly how it works, with all the implications, somewhere in here